MiniApp/Classes/core/RealMiniApp.swift

```diff
@@ -106,59 +124,58 @@ internal class RealMiniApp {
-        self.miniAppStatus.setDownloadStatus(true, appId: appInfo.id, versionId: appInfo.version.versionId)
-        self.miniAppStatus.setCachedVersion(appInfo.version.versionId, for: appInfo.id)
-        isRequiredPermissionsAllowed(
-            appId: appInfo.id,
-            versionId: appInfo.version.versionId) { (result) in
-            switch result {
-            case .success:
-                    DispatchQueue.main.async {
-                        let miniAppDisplayProtocol = self.displayer.getMiniAppView(miniAppId: appInfo.id,
-                                                                                   versionId: appInfo.version.versionId,
-                                                                                   projectId: self.miniAppClient.environment.projectId,
-                                                                                   miniAppTitle: appInfo.displayName ?? "Mini app",
-                                                                                   queryParams: queryParams,
-                                                                                   hostAppMessageDelegate: messageInterface ?? self,
-                                                                                   adsDisplayer: adsDisplayer)
-                        completionHandler(.success(miniAppDisplayProtocol))
-                    }
-            case .failure(let error):
-                completionHandler(.failure(error))
-            }
+    func getMiniAppView(appInfo: MiniAppInfo, manifest: MiniAppManifest?, queryParams: String? = nil, completionHandler: @escaping (Result<MiniAppDisplayDelegate, Error>) -> Void, messageInterface: MiniAppMessageDelegate? = nil, adsDisplayer: MiniAppAdDisplayer? = nil) {
+        DispatchQueue.main.async {
+            let miniAppDisplayProtocol = self.displayer.getMiniAppView(miniAppId: appInfo.id,
+                                                                       versionId: appInfo.version.versionId,
+                                                                       projectId: self.miniAppClient.environment.projectId,
+                                                                       miniAppTitle: appInfo.displayName ?? "Mini app",
+                                                                       manifest: manifest,
+                                                                       queryParams: queryParams,
+                                                                       hostAppMessageDelegate: messageInterface ?? self,
+                                                                       adsDisplayer: adsDisplayer)
+            self.miniAppStatus.setDownloadStatus(true, appId: appInfo.id, versionId: appInfo.version.versionId)
+            self.miniAppStatus.setCachedVersion(appInfo.version.versionId, for: appInfo.id)
+            completionHandler(.success(miniAppDisplayProtocol))
```

Comment:
isRequiredPermissionsAllowed() is removed here. We need to verify if all required permissions are granted before returning the view.

Reply:
sorry, I don't know what happened, a bad merge? please help me to solve this case 😔

Reply:
@Climbatize The Access Token permission changes were merged to the `feature` branch. So actually I think you should have been working on top of that branch instead of `master`. 

But now that we've already merged the meta-data feature to `candidate`, I think we can work on `master` for the access token feature. So probably first we need to merge `feature` to `master` and then you can rebase your branch on `master`.